subprojects/core/src/test/groovy/org/gradle/api/internal/tasks/properties/DefaultPropertyWalkerTest.groovy

```diff
@@ -111,6 +111,17 @@ class DefaultPropertyWalkerTest extends AbstractProjectBuilderSpec {
         1 * visitor.visitInputProperty({ it.propertyName == 'bean' && it.value == null })
     }
 
+    @Unroll
+    def "correct implementation class for #type is discovered"() {
```

Comment:
A little more explanation for this would make sense here.

Reply:
I changed the test name.